contracts/vaults/RibbonThetaVault.sol

```diff
@@ -286,89 +268,66 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
      * @notice Deposits ETH into the contract and mint vault shares. Reverts if the underlying is not WETH.
      */
     function depositETH() external payable nonReentrant {
-        require(asset == WETH, "Not WETH");
-        require(msg.value > 0, "No value");
+        require(asset == WETH, "!WETH");
+        require(msg.value > 0, "!value");
 
-        IWETH(WETH).deposit{value: msg.value}();
         _deposit(msg.value);
+
+        IWETH(WETH).deposit{value: msg.value}();
     }
 
     /**
      * @notice Deposits the `asset` into the contract and mint vault shares.
      * @param amount is the amount of `asset` to deposit
      */
     function deposit(uint256 amount) external nonReentrant {
-        IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
         _deposit(amount);
+
+        IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
     }
 
     /**
      * @notice Mints the vault shares to the msg.sender
      * @param amount is the amount of `asset` deposited
      */
     function _deposit(uint256 amount) private {
-        uint256 totalWithDepositedAmount = totalBalance();
+        uint16 _round = round;
+        uint256 totalWithDepositedAmount = totalBalance().add(amount);
+
+        require(amount < MAX_UINT128, "Overflow");
         require(totalWithDepositedAmount < cap, "Exceed cap");
         require(
             totalWithDepositedAmount >= minimumSupply,
             "Insufficient balance"
         );
 
-        // amount needs to be subtracted from totalBalance because it has already been
-        // added to it from either IWETH.deposit and IERC20.safeTransferFrom
-        uint256 total = totalWithDepositedAmount.sub(amount);
-
-        uint256 shareSupply = totalSupply();
-
-        // Following the pool share calculation from Alpha Homora:
-        // solhint-disable-next-line
-        // https://github.com/AlphaFinanceLab/alphahomora/blob/340653c8ac1e9b4f23d5b81e61307bf7d02a26e8/contracts/5/Bank.sol#L104
-        uint256 share =
-            shareSupply == 0 ? amount : amount.mul(shareSupply).div(total);
-
-        require(shareSupply.add(share) >= minimumSupply, "Insufficient supply");
-
-        emit Deposit(msg.sender, amount, share);
-
-        _mint(msg.sender, share);
```

Comment:
so are [deposits](https://www.notion.so/ribbonfinance/Updated-Vault-Design-29df965a02b7455e9f6b18cb22cfbcf7#109b56f4ce394da6b3ffc9af6b0a04b7) not minting shares?

its either custodial OR non-custodial unlock + withdraw burn?

Reply:
Yep, that's right, deposits no longer mint shares